### ISSUE TITLE
How to go to next step with dynamic dom elements?

### ISSUE THREAD

**Issue by hameedshk** [kamranahmedse/driver.js#279]
Assume there are two text boxes ,tbx1 and tbx2.

1. tbx1 is already active with id on html dom but tbx2 is not yet appeared in dom.
2. Based on the input of tbx1,tbx2 will be enabled in dom.
3. But in driver.js we define the steps with id before the element appears.

Expectation : how to handle dynamic elements in the form of tour. Explain the user tbx1 what it does and once the value is filled,tbx2 appears and need to show the user some info.

This is not an issue but I am looking how to achieve it.And appreciate for the library,thank you

**Comment by nextchamp-saqib** [comment]
Try running `driver.defineSteps()` after the input, it'll redefine the steps according to rendered elements. Then you will have to start the driver again at the last position.

**Comment by hameedshk** [comment]
I have tried this , from first step it goes to second page . Then after first control navigation it stops. No error in console whatsoever

**Comment by bktan81** [comment]
This is a bit hacky but it works for me. I'm using Vuejs and with tons of dynamic elements. I do not like the idea of using nested or multiple driver guides.

Basically, I built a proxy class to wrap the variable "steps" and only evaluate the elements when it's needed. With this, you do not need to call "defineSteps", just call the method "driver.start():.

```      
      var stepsProxy = new Proxy( [], {
        'get': ( target, prop, receiver ) =>{
          if(prop=='length'){
            return stepDefinitions.length-1;
          }

          try{
            let index = parseInt(prop);
            if(!isNaN(index)){
              return this.driver.prepareElementFromStep(stepDefinitions[index], stepDefinitions, index);
            }
          }catch(ex){
            console.log(ex);
          }

          return null;
        }
      });


    //this.driver.defineSteps(steps)
    this.driver.steps = stepsProxy;
    this.driver.start()


```

**Comment by zcf0508** [comment]
#344

**Comment by kamranahmedse** [comment]
We just released [v1.0](https://github.com/kamranahmedse/driver.js/releases/tag/1.0.3) of driver.js which has been rewritten from the ground up. This has a better support for dynamic/asynchronous steps. Please have a [look at the docs](https://driverjs.com/) and do let me know if you face any issues.
